src/main/java/org/manifold/compiler/front/ExpressionEdge.java

```diff
@@ -30,8 +33,7 @@ public void setTarget(ExpressionVertex newTarget) {
 
   public ExpressionEdge(ExpressionVertex from, ExpressionVertex to) {
     if (from == null && to == null) {
-      throw new UndefinedBehaviourError(
-          "attempt to create a totally disconnected ExpressionEdge");
+      log.warn("attempt to create a totally disconnected ExpressionEdge");
```

Comment:
You might have to add checks for null/preconditions where `getSource()` is used in some places  - code may do things like use them as a key in a hash map, causing bugs that are hard to track down.